src/config.js

```diff
@@ -3,6 +3,10 @@ export default class Config {
     return process.env.TRELLIS_BASE_URL || 'http://platform:8080'
   }
 
+  static get platformRootIdentifier() {
```

Comment:
This is the change I like the least. I'd particularly welcome feedback on:

* The name of this config value
* The name of the ENV variable underlying the value
* The default value of `'root_node'`

What is this thing, you might ask? The changes in this branch take Trellis URLs like `http://foo.bar:8080/baz-quux` and indexes them into ES as *only* the path.

Why? So that if we have code around resource templates that is only aware of the RT IDs (the URN-like thingies), it can ask ES if an RT it knows about exists *without* needing knowledge of the Trellis URL.

OK, so we have a predictable mapping from things in Trellis to things in ES, great. But there's an edge case: the root Trellis node. Whenever Trellis emits an event for the root node `http://platform:8080/`, such as when a new top-level container is created or updated, the pipeline will create a new ES entry for... an empty string, not a predictable identifier. The ES server as a result of this will create a new, opaque, internal ID for this node. Which makes predictable edits/updates problematic.

Reply:
how about baseUrl or sinopiaServerBaseUrl or trellisBaseUrl (snake case, whatever) instead of root ? 

as for the BaseContainer of Trellis ... that is worrying.   I *think* we need that BaseContainer id to be stable.   Hrmmm.  Maybe John or Jeremy has a good idea about that.

Reply:
hmm... what if the identifier always contains the leading slash instead of removing it?  would ES barf on a leading slash, or only a slash?  that would seem to solve the problem, because then the root node is just always `/`.

if that would be problematic, i dunno... just something that's really unlikely to be used, or an invalid URI but a valid ES identifier?  `___MY1337ROOTNODEID___`?

i'm fine w/ the names as-is.

Reply:
ok, actually, a naming suggestion if this alternate identifier is needed:  `rootNodeIdentifier` for the method name, and `ROOT_NODE_IDENTIFIER` for the env var name.

@ndushay said:
> as for the BaseContainer of Trellis ... that is worrying. I think we need that BaseContainer id to be stable.

which triggers two thoughts:
1) i think from trellis' perspective, it is stable, it's just elastic search that will fill in some opaque internal ID, right?  and actually, since we're using `/repository` as the base from trellis' point of view...
2) maybe the corner case won't be an issue IRL, at least as we've designed the interactions w/ trellis?  which is to say, we don't expect to be updating `/`, i don't think, because we only expect to deal with `/repository` and things underneath that.  that said... i think our indexer should be prepared to handle updates to `/`, so i am all for addressing this corner case.

Reply:
> ok, actually, a naming suggestion if this alternate identifier is needed: `rootNodeIdentifier` for the method name, and `ROOT_NODE_IDENTIFIER` for the env var name.

I dig this.

> 1. i think from trellis' perspective, it is stable, it's just elastic search that will fill in some opaque internal ID, right?  and actually, since we're using `/repository` as the base from trellis' point of view...

Yes, that is true. The concerns here are all on the ES side.

> 2. maybe the corner case won't be an issue IRL, at least as we've designed the interactions w/ trellis?  which is to say, we don't expect to be updating `/`, i don't think, because we only expect to deal with `/repository` and things underneath that.  that said... i think our indexer should be prepared to handle updates to `/`, so i am all for addressing this corner case.

Right, this is def a corner case. OK, I'll make the changes you suggest. Thanks!

Reply:
> hmm... what if the identifier always contains the leading slash instead of removing it? would ES barf on a leading slash, or only a slash? that would seem to solve the problem, because then the root node is just always `/`.

It wouldn't barf, no, but the slash would be escaped so we'd be in a situation where we're telling ES that the identifier is `/` and getting it via `%2F`, which could be confusing over time.
 
> if that would be problematic, i dunno... just something that's really unlikely to be used, or an invalid URI but a valid ES identifier? `___MY1337ROOTNODEID___`?

I like the idea of using leading and trailing underscores, which are valid ES identifier characters. I'm going to pursue that. I consider it unlikely that we're going to create another top-level container with leading and trailing underscores, so I am unconcerned about the likelihood of a conflict with this special case. Thanks!

Reply:
👍